yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -544,6 +545,7 @@ private[spark] class Client(
       logInfo(s"Credentials file set to: $credentialsFile")
       val renewalInterval = getTokenRenewalInterval(stagingDirPath)
       sparkConf.set("spark.yarn.token.renewal.interval", renewalInterval.toString)
+      SparkHadoopUtil.get.startExecutorDelegationTokenRenewer(sparkConf)
```

Comment:
This depends on the lines immediately above this though. `setupCredentials` gets called way before this I think.